applications/electron/scripts/flatpak-helpers.js

```diff
@@ -0,0 +1,194 @@
+            '--host',
+            'flatpak',
+            'info', 
+            '--show-location',
+            flatpakId
+        ], { encoding: 'utf8' });
+
+        const installDir = result.trim();
+        // The actual files are in the 'files' subdirectory of the installation
+        return path.join(installDir, 'files');
+    } catch (error) {
+        console.error('Failed to get Flatpak installation directory:', error.message);
+        return null;
+    }
+}
+
+/**
+ * Gets the arguments to forward XDG environment variables to the host.
+ * These are prefixed with FLATPAK_ to avoid conflicts.
+ * @returns {string[]}
+ */
+function getXdgEnvArgs() {
+    const args = [];
+    for (const key of XDG_KEYS) {
+        const value = process.env[key];
+        if (value) {
+            args.push(`--env=FLATPAK_${key}=${value}`);
+        }
+    }
+    return args;
+}
+
+/**
+ * Attempts to restart the application outside the Flatpak sandbox.
+ * 
+ * If running inside a Flatpak sandbox, this function will:
+ * 1. Determine the Flatpak installation directory
+ * 2. Build the command to restart on the host using flatpak-spawn
+ * 3. Execute the command synchronously, then exit with its code
+ * 
+ * If successful, this function does not return (process.exit is called).
+ * If not in a Flatpak or escape fails, this function returns normally.
+ */
+function tryRestartToHost() {
+    const flatpakDir = getFlatpakDir();
+    if (!flatpakDir) {
+        // Not in Flatpak or escape disabled - add any extra libs and continue normally
+        addFlatpakLibsToPath();
+        return;
+    }
+
+    console.log('Detected Flatpak sandbox, restarting on host system...');
+
+    // Build the flatpak-spawn command
+    const args = [
+        '--host'
+    ];
+
+    // Forward XDG environment variables
+    args.push(...getXdgEnvArgs());
+
+    // Set environment variable to indicate updates should use Flatpak
+    args.push('--env=THEIA_UPDATE_EXPLANATION=Please use Flatpak to update Theia IDE');
+
+    // Pass the path to bundled libraries for LD_LIBRARY_PATH on host
+    const libPath = path.join(flatpakDir, 'lib');
+    args.push(`--env=${EXTRA_LIB_ENV_NAME}=${libPath}`);
+
+    // Prevent the restarted process from trying to escape again
+    args.push(`--env=${NO_ESCAPE_ENV_NAME}=1`);
+
+    // The wrapper script that launches the Electron app
+    // This should be the 'theia' command installed by the Flatpak
+    const theiaCommand = path.join(flatpakDir, 'bin', 'theia');
+    args.push(theiaCommand);
+
+    // Forward all command line arguments
+    // In Electron, process.argv[0] is the electron binary, process.argv[1] may be the main script
+    // We want to forward user-provided arguments
+    const userArgs = process.argv.slice(2);
+    args.push(...userArgs);
+
+    try {
+        // Use spawnSync to run synchronously and inherit stdio
+        const result = spawnSync('/usr/bin/flatpak-spawn', args, {
```

Comment:
I think that `execve`, where available, is a cleaner approach and it's easier on the system. See [my attempt at it](https://github.com/livingsilver94/theia/commit/cfee17e99579b4c2c2b6bd882ebfbbac422a33cc#diff-90173aa330880b1568e7b48b178180a7871be970f1a563bbae79bf042a06dbc6R25).

Reply:
this file was dropped entirely